pysindy/_core.py

```diff
@@ -847,6 +878,43 @@ def fit(
 
         return self
 
+    def equations(self, precision: int = 3) -> list[str]:
```

Comment:
Another option to overriding this function would be to override how `self.feature_names` is set in `self._fit_shape()`.  You could even leave that alone and make `self.feature_names` a `@property`, defining different setting and getting behavior.  See [property](https://docs.python.org/3/library/functions.html#property).

No change required here, just a thought, since I realized there's actually a lot of repeated code from super().equations().